Include/pymacro.h

```diff
@@ -10,6 +10,12 @@
 #  define static_assert _Static_assert
 #endif
 
+// static_assert isd efined in GLIB from version 2.16. Before it requires
+// compiler support (gcc >= 4.6) and is called _Static_assert.
+#if defined(__GLIBC__) && __GLIBC__ <= 2 && __GLIBC_MINOR__ <= 16
```

Comment:
Maybe don't redefine static_assert if it's already defined, just in case? I adjusted the test for glib 1.x with x > 16 (I don't think that it exists in practice :-)). I let you reindent the code to limit to 80 columns:

```suggestion
#if defined(__GLIBC__) && (__GLIBC__ < 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ <= 16) && !defined(static_assert)
```